src/compiler.ts

```diff
@@ -5642,9 +5673,26 @@ export class Compiler extends DiagnosticEmitter {
           }
           return this.module.unreachable();
         }
+        if (indexedSet.is(CommonFlags.STATIC)) {
+          this.error(
+            DiagnosticCode._0_modifier_cannot_be_used_here,
+            indexedSet.identifierNode.range, CommonNames.static_
+          );
+          return this.module.unreachable();
+        }
```

Comment:
I *guess* this could theoretically be allowed, similar to other overloads. Did this error before?

Reply:
Static overloaded indexed sets/gets doesn't really make sense. Static variant, usually useful when type of lhs != rhs and rhs should be an instance of class. For example: 
```ts
class Foo {
  @overload("-")
  static sub(lhs: i32, rhs: Foo): Foo  { ... }
}
```
But static indexed sets/gets will always required instance of class as first argument:
```ts
class Foo<T> {
  @overload("[]=")
  static _set(self: Foo, index: i32, value: T): this  { ... }
  //          ^^^^^^^^^
}
```

And finally, in theory we can add class based indeed sets / gets which may conflict with this. I mean:
```ts
// this doesn't support

class Foo {
  static inner: i32 = 0;
  @overload("[]=")
  static _set(index: i32, value: i32): this {
     Foo.inner = value;
  }
}

Foo[3] = 124;
```

Reply:
Can't fully follow the line of arguments. Iirc the first sample isn't supported generally, and the second and third seem to be the same for most overloads? Mostly interested in making this somewhat coherent at least. I guess static indexed overloads didn't work before as well?